helix-core/src/main/java/org/apache/helix/HelixPropertyFactory.java

```diff
@@ -113,10 +109,10 @@ public static CloudConfig getCloudConfig(String zkAddress, String clusterName) {
       // The try-catch logic is for backward compatibility reason only. Even if the cluster is not set
       // up yet, constructing a new ZKHelixManager should not throw an exception
       try {
-        cloudConfig = configAccessor.getCloudConfig(clusterName) == null ? buildEmptyCloudConfig()
+        cloudConfig = configAccessor.getCloudConfig(clusterName) == null ? new CloudConfig()
             : configAccessor.getCloudConfig(clusterName);
```

Comment:
This queries cloudConfig twice, and not atomic... Can we assign it with `configAccessor.getCloudConfig(clusterName)` first and then do null check? 
(sorry I missed this one in the first pass)

Reply:
Updated.